### PR TITLE
Geopy 1014

### PR DESCRIPTION
**Pull request by MatthieuCMira**
**GEOPY-1014 - Crop geoimage by extent yields large files. Look into compression format of PIL**
geoimage compression.
Geoimage is now a png file.

### CODE REVIEW

File: geoh5py/objects/geo_image.py [hunk 1]
```diff
@@ -380,12 +382,16 @@ def image(self, image: str | np.ndarray | BytesIO | Image.Image):
                 f"Get type {type(image)} instead."
             )
 
-        with TemporaryDirectory() as tempdir:
-            ext = getattr(image, "format")
-            temp_file = (
-                Path(tempdir) / f"image.{ext.lower() if ext is not None else 'tiff'}"
+        if image.mode not in ["RGBA", "RGB", "L"]:
```
Comment (domfournier) [inline geoh5py/objects/geo_image.py:385]: I think "CMYK" would be supported as well

File: geoh5py/shared/conversion/grid2d.py [hunk 2]
```diff
@@ -211,6 +211,9 @@ def convert_to_pillow(data: np.ndarray, mode: str | None = None) -> Image:
             if mode is None:
                 raise IndexError("Only 1, 3, or 4 layers can be selected")
 
+        if mode not in ["L", "RGB", "RGBA"]:
```
Comment (domfournier) [inline geoh5py/shared/conversion/grid2d.py:214]: Same here.

**Review by domfournier — COMMENTED** [review]
See comment about CMYK

File: geoh5py/objects/geo_image.py (hunk 1 above)
Comment (MatthieuCMira) [reply geoh5py/objects/geo_image.py:385]: No it won't. 
PNG files does not support it. 

A workaround could be to manage the creation of a jpeg if mode = "CMYK".
However, it takes other compression options, and that's more code. 

Which option do we choose?

File: geoh5py/shared/conversion/grid2d.py (hunk 2 above)
Comment (MatthieuCMira) [reply geoh5py/shared/conversion/grid2d.py:214]: Same answer

File: geoh5py/objects/geo_image.py (hunk 1 above)
Comment (MatthieuCMira) [reply geoh5py/objects/geo_image.py:385]: Create a dictionary to manage compression options

File: geoh5py/shared/conversion/grid2d.py [hunk 3]
```diff
@@ -207,10 +211,16 @@ def convert_to_pillow(data: np.ndarray, mode: str | None = None) -> Image:
                 mode = "RGB"
             elif data.shape[-1] == 4:
                 mode = "RGBA"
+                # todo: I do not like RGBA as default, rather CMYK.
+                #  crop image by extent should return the same format,
+                #  not RGBA, and 0 for nan values.
```
Comment (domfournier) [inline geoh5py/shared/conversion/grid2d.py:216]: I don't agree, but we should have that conversation. Remove comment here.

**Pull request merged**